crates/base_db/src/input.rs

```diff
@@ -150,11 +150,20 @@ pub enum ProcMacroKind {
     Attr,
 }
 
+pub trait ProcMacroExpander: fmt::Debug + Send + Sync + RefUnwindSafe {
+    fn expand(
+        &self,
+        subtree: &Subtree,
+        attrs: Option<&Subtree>,
+        env: &Env,
```

Comment:
Hm, yeah, I think this makes sense -- proc macro can observe compile-time env variables.